.github/workflows/detect_changes.yml

```diff
@@ -0,0 +1,34 @@
+name: Detect Dependencies Changes
+
+on:
+  pull_request:
+    types: [opened, synchronize]
+
+jobs:
+  update-deps:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v1
+        with:
+          python-version: '3.8.x'
+      - name: Install Dependencies
+        run: |
+          pip install virtualenv
+          virtualenv test_python -q
+          source test_python/bin/activate
+          make installdeps
+      - name: Detect Dependency Changes
+        id: check
+        run: |
+          source test_python/bin/activate
+          mkdir /tmp/dependencies_updated_artifacts
+          export DEPENDENCY_FILE_PATH=/tmp/dependencies_updated_artifacts/current_dependencies.txt
+          evalml/tests/dependency_update_check/make_deps_diff.sh
+          diff evalml/tests/dependency_update_check/latest_dependency_versions.txt /tmp/dependencies_updated_artifacts/current_dependencies.txt > /tmp/dependencies_updated_artifacts/diff.txt
+          exit $?
```

Comment:
@ParthivNaresh will GH actions error out if this error code is nonzero? That was the case with circleci, just confirming that'll work for GH too. If not, we should change this to make sure we get an error when there is a non-empty diff.

Reply:
Here's the result of the updated yaml:
CI Workflow: https://app.circleci.com/pipelines/github/alteryx/evalml/10515/workflows/014d87b3-38cb-4ee1-83b2-c7885c68f52a/jobs/120417
GitHub Action: https://github.com/alteryx/evalml/runs/2058345037?check_suite_focus=true